Modules/Quotas/src/Quotas.Infrastructure/Persistence/Repository/IdentitiesRepository.cs

```diff
@@ -27,9 +28,17 @@ public async Task Add(Identity identity, CancellationToken cancellationToken)
         await _dbContext.SaveChangesAsync(cancellationToken);
     }
 
-    public async Task<Identity> FindById(string identityId, CancellationToken cancellationToken)
+    public async Task<Identity> FindById(string identityAddress, CancellationToken cancellationToken)
     {
-        return await _identitiesDbSet.FindAsync(identityId, cancellationToken);
+        return await _identitiesDbSet.FirstOrDefaultAsync(id => id.Address == identityAddress, cancellationToken) ?? throw new NotFoundException(identityAddress);
```

Comment:
`id` is a really confusing name.

Further, the NotFoundException expects the name of the entity that wasn't found as a parameter, and not the Id that wasn't found.

```suggestion
        return await _identitiesDbSet.FirstOrDefaultAsync(i => i.Address == identityAddress, cancellationToken) ?? throw new NotFoundException(identityAddress);
```